src/pug/parser.ts

```diff
@@ -0,0 +1,64 @@
+import type { Token } from "pug-lexer";
```

Comment:
I would like to try to return the original Tokens from pug, so the plugin can easily use them.
I typed these really well (https://github.com/pugjs/pug/blob/master/packages/pug-lexer/index.d.ts)